### ISSUE TITLE
XPath queries with xmllint 

### ISSUE THREAD

**Issue by infojunkie** [aautio/validate-with-xmllint#5]
Thanks for a great module! I'm happily using it to validate XML generated by [my MusicXML converter](https://github.com/infojunkie/ireal-musicxml/blob/main/test/musicxml.spec.js).

One additional feature that would be great to have here is to also run XPath queries using `xmllint`. Right now, I am using a combination of `xpath.js` and `xmldom` but having this functionality inside your module would be great to reduce dependencies and use a solid tool for this work. Thanks!

**Comment by infojunkie** [comment]
Closing this, after finding significant shortcomings with `xmllint --xpath`. Sorry for the noise!

**Comment by aautio** [comment]
@infojunkie sorry I had no time to respond earlier.

I really liked your idea! Using xpath selection with xmllint would help my workflows as well.

Out of curiosity: What kinds of shortcomings did you run into with it?

**Comment by infojunkie** [comment]
Some of the shortcomings I encountered while converting my tests to use `xmllint --xpath`:

1. Difficult to separate multiple results from each other, especially when each result spans multiple stdout line - cf. [this PR](https://gitlab.gnome.org/GNOME/libxml2/-/merge_requests/8). The outcome is that each stdout line is considered a separate result. 

2. Difficult or no way to convince `xmllint` to return attribute _values_ only, as opposed to `attribute="value"` format - in the case of multiple results.

Happy to share with you the code I made to add a `queryXPath` function to your repo. It worked correctly (with tests), except for the limitations above.
